packages/material-ui-lab/src/PickersDay/PickersDay.tsx

```diff
@@ -248,38 +303,39 @@ const PickersDay = React.forwardRef(function PickersDay<TDate>(
     }
   }
 
-  const dayClassName = clsx(
-    classes.root,
-    {
-      [classes.selected]: selected,
-      [classes.dayWithMargin]: !disableMargin,
-      [classes.today]: !disableHighlightToday && isToday,
-      [classes.dayOutsideMonth]: outsideCurrentMonth && showDaysOutsideCurrentMonth,
-    },
-    className,
-  );
+  if (outsideCurrentMonth && !showDaysOutsideCurrentMonth) {
+    return <div aria-hidden className={clsx(classes.root, className)} />;
+  }
+
+  let injectedProps: Parameters<typeof PickersDayRoot>[0] & { 'data-mui-test'?: string } = {
+    styleProps,
+    className: clsx(classes.root, className),
+    children: null,
+  };
 
   if (outsideCurrentMonth && !showDaysOutsideCurrentMonth) {
-    return <div aria-hidden className={clsx(dayClassName, classes.hiddenDaySpacingFiller)} />;
+    injectedProps = {
```

Comment:
This should be a different Root component, see that this is `div` not `ButtonBase`, we cannot reuse the same element.